python/dgllife/utils/complex_to_graph.py

```diff
@@ -56,6 +58,148 @@ def get_atomic_numbers(mol, indices):
+
+    assert ligand_coordinates is not None, 'Expect ligand_coordinates to be provided.'
+    assert protein_coordinates is not None, 'Expect protein_coordinates to be provided.'
+    if max_num_ligand_atoms is not None:
+        assert max_num_ligand_atoms >= ligand_mol.GetNumAtoms(), \
+            'Expect max_num_ligand_atoms to be no smaller than ligand_mol.GetNumAtoms(), ' \
+            'got {:d} and {:d}'.format(max_num_ligand_atoms, ligand_mol.GetNumAtoms())
+    if max_num_protein_atoms is not None:
+        assert max_num_protein_atoms >= protein_mol.GetNumAtoms(), \
+            'Expect max_num_protein_atoms to be no smaller than protein_mol.GetNumAtoms(), ' \
+            'got {:d} and {:d}'.format(max_num_protein_atoms, protein_mol.GetNumAtoms())
+
+    if strip_hydrogens:
+        # Remove hydrogen atoms and their corresponding coordinates
+        ligand_atom_indices_left = filter_out_hydrogens(ligand_mol)
+        protein_atom_indices_left = filter_out_hydrogens(protein_mol)
+        ligand_coordinates = ligand_coordinates.take(ligand_atom_indices_left, axis=0)
+        protein_coordinates = protein_coordinates.take(protein_atom_indices_left, axis=0)
+    else:
+        ligand_atom_indices_left = list(range(ligand_mol.GetNumAtoms()))
+        protein_atom_indices_left = list(range(protein_mol.GetNumAtoms()))
+
+    # Compute number of nodes for each type
+    if max_num_ligand_atoms is None:
+        num_ligand_atoms = len(ligand_atom_indices_left)
+    else:
+        num_ligand_atoms = max_num_ligand_atoms
+
+    if max_num_protein_atoms is None:
+        num_protein_atoms = len(protein_atom_indices_left)
+    else:
+        num_protein_atoms = max_num_protein_atoms
+
+    # Construct bigraph for stage 1
+    node_featurizer = CanonicalAtomFeaturizer(atom_data_field='h')
+    edge_featurizer = CanonicalBondFeaturizer(bond_data_field='e')
+    ligand_bigraph = mol_to_bigraph(ligand_mol, add_self_loop=False,
+                                    node_featurizer=node_featurizer,
+                                    edge_featurizer=edge_featurizer,
+                                    canonical_atom_order=False) # Keep the original atomic order)
+    protein_bigraph = mol_to_bigraph(protein_mol, add_self_loop=False,
+                                     node_featurizer=node_featurizer,
+                                     edge_featurizer=edge_featurizer,
+                                     canonical_atom_order=False)
+
+    complex_bigraph = batch([ligand_bigraph, protein_bigraph])
+    # remove features that never appear
+    zero_h_cols = [5, 13, 14, 16, 17, 19, 20, 21, 22, 23, 24, 27, 30, 31, 33, 36, 38, 39, 40, 42, 50, 51, 52, 53, 58, 59, 60, 62, 63, 64, 65, 66, 67, 73]
+    zero_e_cols = [4,  5,  7,  8,  9, 10, 11]
+    complex_bigraph.ndata['h'] = np.delete(complex_bigraph.ndata['h'], zero_h_cols, axis=1)
+    complex_bigraph.edata['e'] = np.delete(complex_bigraph.edata['e'], zero_e_cols, axis=1) # 5 edge types remain
```

Comment:
If this is the case, you should define custom featurizers rather than use `CanonicalAtomFeaturizer` and `CanonicalBondFeaturizer` and then remove them.